src/NUnitEngine/nunit.engine.core/Drivers/NUnitFrameworkApi2018.cs

```diff
@@ -0,0 +1,293 @@
+    /// may be used for NUnit 3.10 or higher.
+    /// </summary>
+#if NETFRAMEWORK
+    public class NUnitFrameworkApi2018 : MarshalByRefObject, NUnitFrameworkApi
+#else
+    public class NUnitFrameworkApi2018 : NUnitFrameworkApi
+#endif
+    {
+        static readonly ILogger log = InternalTrace.GetLogger(nameof(NUnitFrameworkApi2018));
+
+        const string LOAD_MESSAGE = "Method called without calling Load first. Possible error in runner.";
+        const string INVALID_FRAMEWORK_MESSAGE = "Running tests against this version of the framework using this driver is not supported. Please update NUnit.Framework to the latest version.";
+        const string FAILED_TO_LOAD_ASSEMBLY = "Failed to load assembly ";
+        const string FAILED_TO_LOAD_NUNIT = "Failed to load the NUnit Framework in the test assembly";
+
+        const string CONTROLLER_TYPE = "NUnit.Framework.Api.FrameworkController";
+
+        private string _driverId;
+
+        AssemblyName _nunitRef;
+        string? _testAssemblyPath;
+
+        object? _frameworkController;
+        Type? _frameworkControllerType;
+
+
+        public NUnitFrameworkApi2018(string driverId, AssemblyName nunitRef)
+        {
+            Guard.ArgumentNotNull(driverId, nameof(driverId));
+            Guard.ArgumentNotNull(nunitRef, nameof(nunitRef));
+
+            _driverId = driverId;
+#if NETFRAMEWORK
+            //if (RemotingServices.IsTransparentProxy(driver))
+            //{
+            //    // We need to replace nunitRef with a reference valid in the current domain.
+            //    var frameworkRef = AssemblyNameReference.Parse(nunitRef.FullName);
+            //    var frameworkDef = new DefaultAssemblyResolver().Resolve(frameworkRef);
+            //    frameworkDef.MainModule.
+            //}
+            //if (_nunitRef == null)
+                _nunitRef = nunitRef;
+#else
+            _nunitRef = nunitRef;
+#endif
+        }
+
+#if NETCOREAPP
+        TestAssemblyLoadContext? _assemblyLoadContext;
+        Assembly? _testAssembly;
+        Assembly? _frameworkAssembly;
+#endif
+
+        public string Load(string testAssemblyPath, IDictionary<string, object> settings)
+        {
+            Guard.ArgumentNotNull(testAssemblyPath, nameof(testAssemblyPath));
+            Guard.ArgumentNotNull(settings, nameof(settings));
+            Guard.ArgumentValid(File.Exists(testAssemblyPath), "Framework driver called with a file name that doesn't exist.", "testAssemblyPath");
+            log.Info($"Loading {testAssemblyPath} - see separate log file");
+
+            _testAssemblyPath = Path.GetFullPath(testAssemblyPath);
+            var idPrefix = string.IsNullOrEmpty(_driverId) ? "" : _driverId + "-";
+
+#if NETFRAMEWORK
+            // Normally, the caller should check for an invalid requested runtime, but we make sure here
+            var requestedRuntime = settings.ContainsKey(EnginePackageSettings.RequestedRuntimeFramework)
```

Comment:
_requestedRuntime_ is not used here. I would think it is too late then, it should be selected to before to run this function.

Reply:
The sole purpose is to ensure that the error message thrown makes sense to the user. That is, if the user requested a particular runtime, it's a user error. If it's due to the engine picking the wrong runtime, it's an internal error. I'll finish some others and come back to this.

Reply:
Removing this.